kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp

```diff
@@ -171,7 +171,11 @@ knn_search(A& index, B& query, C& k_indices, D& dists, unsigned int k, F& params
   // Wrap indices vector (no data allocation)
   ::flann::Matrix<std::size_t> indices_mat(&indices[0], 1, k);
   auto ret = index.knnSearch(query, indices_mat, dists, k, params);
-  std::copy(indices.cbegin(), indices.cend(), k_indices.begin());
+  // cast appropriately
+  std::transform(indices.cbegin(),
+                 indices.cend(),
+                 k_indices.begin(),
+                 [](std::size_t x) { return static_cast<pcl::index_t>(x); });
```

Comment:
```suggestion
                 [](const auto& x) { return static_cast<pcl::index_t>(x); });
```
Just an idea, probably makes no difference

Reply:
Makes no difference. But I made the change anyways